pulpcore/cli/common/generic.py

```diff
@@ -448,6 +448,65 @@ def pulp_option(*args: Any, **kwargs: Any) -> Callable[[FC], FC]:
     return click.option(*args, **kwargs)
 
 
+def resource_lookup_option(*args: Any, **kwargs: Any) -> Callable[[FC], FC]:
+    lookup_key: str = kwargs.pop("lookup_key", "name")
+    href_pattern: Optional[str] = kwargs.pop("href_pattern", None)
+
+    def _option_callback(
+        ctx: click.Context, param: click.Parameter, value: Optional[str]
+    ) -> EntityFieldDefinition:
+        # Pass None and "" verbatim
+        if not value:
+            return value
+
+        pulp_href: Optional[str] = None
+        entity: Optional[EntityDefinition] = None
+
+        pulp_ctx = ctx.find_object(PulpCLIContext)
+        assert pulp_ctx is not None
+
+        if value.startswith("/"):
+            # An href was passed
+            if href_pattern is None:
+                raise click.ClickException(
+                    _("The option {option_name} does not support href.").format(
+                        option_name=param.name
+                    )
+                )
+
+            pattern = rf"^{pulp_ctx.api_path}{href_pattern}"
+            if not re.match(pattern, value):
+                raise click.ClickException(
+                    _("'{value}' is not a valid href for {option_name}.").format(
+                        value=value, option_name=param.name
+                    )
+                )
+
+            pulp_href = value
+        else:
+            # A natural key identifier was passed
+            identifier = value.split(":", maxsplit=2)[-1]
+            entity = {lookup_key: identifier}
+
+        if entity:
+            ctx.obj.entity = entity
```

Comment:
and use `ctx.find_object(context_class)` here instead. This is the way we find the _right_ entity context and not only the _last_ one. The requirement being that the context class has or inherits an `href_pattern` obviously.